File: server/domain/ports/log.go
// screenreader-mcp domain -- the Log port.
// Copyright (C) 2026 Marlon Brandao de Sousa. GPL-2. See COPYING.txt.
//
// ROLE: domain port, diagnostics for the domain and the adapters.
// IMPLEMENTED BY: adapters/stderr_log.go, and fakes/log.go in tests.
// USED BY: the bridge client and the connection controller.
//
// Stdout carries MCP frames only; one stray print corrupts the JSON-RPC stream.
package ports

type Log interface {
	Debugf(format string, args ...any)

	Infof(format string, args ...any)

	Errorf(format string, args ...any)
}

File: bridges/nvda/addon/globalPlugins/nvdaMcpBridge/domain/ports/log.py
# nvdaMcpBridge domain ports -- the Log port.
# Copyright (C) 2026 Marlon Brandao de Sousa. GPL-2. See COPYING.txt.
# ROLE: domain port, the logging seam that keeps adapters testable without NVDA's ``logHandler``.
# IMPLEMENTED BY: adapters/nvda_log.py; tests/fakes/log.py.
# USED BY: adapters/ini_bridge_config.py.

from __future__ import annotations

from abc import ABC, abstractmethod


class Log(ABC):
	@abstractmethod
	def info(self, msg: str) -> None: ...

	@abstractmethod
	def warning(self, msg: str) -> None: ...

	@abstractmethod
	def error(self, msg: str, exc_info: bool = False) -> None: ...
